deploy/charts/cert-manager/templates/startupapicheck-psp.yaml

```diff
@@ -0,0 +1,50 @@
+{{- if .Values.startupapicheck.enabled -}}
+{{- if .Values.global.podSecurityPolicy.enabled }}
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: {{ template "startupapicheck.fullname" . }}
+  labels:
+    app: {{ include "startupapicheck.name" . }}
+    app.kubernetes.io/name: {{ include "startupapicheck.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component: "startupapicheck"
+    {{- include "labels" . | nindent 4 }}
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: 'docker/default'
+    seccomp.security.alpha.kubernetes.io/defaultProfileName:  'docker/default'
+    {{- if .Values.global.podSecurityPolicy.useAppArmor }}
+    apparmor.security.beta.kubernetes.io/allowedProfileNames: 'runtime/default'
+    apparmor.security.beta.kubernetes.io/defaultProfileName:  'runtime/default'
+    {{- end }}
+    {{- if .Values.startupapicheck.rbac.annotations }}
+    {{ toYaml .Values.startupapicheck.rbac.annotations | nindent 4 }}
+    {{- end }}
+spec:
+  privileged: false
+  allowPrivilegeEscalation: false
+  allowedCapabilities: []  # default set of capabilities are implicitly allowed
+  volumes:
+  - 'projected'
```

Comment:
```suggestion
  - 'projected'
  - 'secret'
```

We've been testing this afternoon and found that `secret` is also required, at least on GKE.

```
$ kubectl get pods -n cert-manager cert-manager-startupapicheck-krqxb -o yaml 
...

  volumes:
  - name: cert-manager-startupapicheck-token-6p2rc
    secret:
      defaultMode: 420
      secretName: cert-manager-startupapicheck-token-6p2rc

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.0", GitCommit:"c2b5237ccd9c0f1d600d3072634ca66cefdf272f", GitTreeState:"clean", BuildDate:"2021-08-04T18:03:20Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.8-gke.900", GitCommit:"28ab8501be88ea42e897ca8514d7cd0b436253d9", GitTreeState:"clean", BuildDate:"2021-06-30T09:23:36Z", GoVersion:"go1.15.13b5", Compiler:"gc", Platform:"linux/amd64"}

```


Without the secret volume type we see the following error when the startupapi pod is created

```
$ kubectl -n cert-manager get events
LAST SEEN   TYPE      REASON              OBJECT                                          MESSAGE
29s         Normal    Scheduled           pod/cert-manager-cainjector-7898b8578f-frsd2    Successfully assigned cert-manager/cert-manager-cainjector-7898b8578f-frsd2 to gke-sample-cluster-default-pool-ea6fd01d-t0j3
28s         Normal    Pulled              pod/cert-manager-cainjector-7898b8578f-frsd2    Container image "quay.io/jetstack/cert-manager-cainjector:v1.5.2" already present on machine
28s         Normal    Created             pod/cert-manager-cainjector-7898b8578f-frsd2    Created container cert-manager
27s         Normal    Started             pod/cert-manager-cainjector-7898b8578f-frsd2    Started container cert-manager
30s         Normal    SuccessfulCreate    replicaset/cert-manager-cainjector-7898b8578f   Created pod: cert-manager-cainjector-7898b8578f-frsd2
30s         Normal    ScalingReplicaSet   deployment/cert-manager-cainjector              Scaled up replica set cert-manager-cainjector-7898b8578f to 1
30s         Normal    Scheduled           pod/cert-manager-cdd857fb4-7m6t9                Successfully assigned cert-manager/cert-manager-cdd857fb4-7m6t9 to gke-sample-cluster-default-pool-ea6fd01d-t0j3
28s         Normal    Pulled              pod/cert-manager-cdd857fb4-7m6t9                Container image "quay.io/jetstack/cert-manager-controller:v1.5.2" already present on machine
28s         Normal    Created             pod/cert-manager-cdd857fb4-7m6t9                Created container cert-manager
28s         Normal    Started             pod/cert-manager-cdd857fb4-7m6t9                Started container cert-manager
30s         Normal    SuccessfulCreate    replicaset/cert-manager-cdd857fb4               Created pod: cert-manager-cdd857fb4-7m6t9
5s          Warning   FailedCreate        job/cert-manager-startupapicheck                Error creating: pods "cert-manager-startupapicheck-" is forbidden: PodSecurityPolicy: unable to admit pod: [spec.volumes[0]: Invalid value: "secret": secret volumes are not allowed to be used]
29s         Normal    Scheduled           pod/cert-manager-webhook-979969556-tmlg7        Successfully assigned cert-manager/cert-manager-webhook-979969556-tmlg7 to gke-sample-cluster-default-pool-ea6fd01d-t0j3
28s         Normal    Pulled              pod/cert-manager-webhook-979969556-tmlg7        Container image "quay.io/jetstack/cert-manager-webhook:v1.5.2" already present on machine
28s         Normal    Created             pod/cert-manager-webhook-979969556-tmlg7        Created container cert-manager
27s         Normal    Started             pod/cert-manager-webhook-979969556-tmlg7        Started container cert-manager
30s         Normal    SuccessfulCreate    replicaset/cert-manager-webhook-979969556       Created pod: cert-manager-webhook-979969556-tmlg7
30s         Normal    ScalingReplicaSet   deployment/cert-manager-webhook                 Scaled up replica set cert-manager-webhook-979969556 to 1
30s         Normal    ScalingReplicaSet   deployment/cert-manager                         Scaled up replica set cert-manager-cdd857fb4 to 1

```

On @irbekrm 's Kubernetes 1.21 cluster using Kind, she noticed that the startupapicheck Pod instead uses a projected volume comprising secret, configmap and downwardAPI components.

So perhaps we need to add both volume types for compatibility with different cluster versions.

Reply:
Looking at [Kubernetes v1.21 release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.21.md):

> The BoundServiceAccountTokenVolume feature has been promoted to beta, and enabled by default

As I understand this means that by default in Kubernetes v1.20 and older the service account token will be mounted as a secret whereas in v1.21 it will be mounted as a projected volume.